crypto/bio/bio.c

```diff
@@ -333,13 +392,16 @@ int BIO_puts(BIO *bio, const char *in) {
     OPENSSL_PUT_ERROR(BIO, BIO_R_UNSUPPORTED_METHOD);
     return -2;
   }
-  if(HAS_CALLBACK(bio)) {
-    long callback_ret = bio->callback_ex(bio, BIO_CB_PUTS, in, 0, 0, 0L, 1L, NULL);
-    if (callback_ret <= 0) {
-      if (callback_ret >= INT_MIN) {
-        return (int)callback_ret;
+  if (HAS_CALLBACK(bio)) {
+    BIO_callback_fn_ex cb = get_callback(bio);
+    if (cb != NULL) {
```

Comment:
Actually, I don't see much value in the `HAS_CALLBACK` macro b/c you're still having to subsequently check `if (cb != NULL) {`.

Why not just eliminate the macro call and just have:
```
  BIO_callback_fn_ex cb = get_callback(bio);
  if (cb != NULL) {
    ...
  }
```